src/constructors.jl

```diff
@@ -75,26 +81,32 @@ for ArbT in (:Arf, :Arb, :Acb, :Mag)
         if $ArbT != Arf
             $ArbT(si::Int64, prec::Integer) = set!($ArbT(prec), si)
         end
+
+        Base.zero(t::$ArbT) = $ArbT(0, precision(t))
+        Base.one(t::$ArbT) = $ArbT(1, precision(t))
     end
 end
 
-Arf(x::BigFloat, prec::Integer) = set!(Arf(prec), x)
-Arb(x::BigFloat, prec::Integer) = set!(Arb(prec), Arf(x, prec))
-Acb(x::BigFloat, prec::Integer) = set!(Acb(prec), Arb(x, prec))
+Arf(x::BigFloat, prec::Integer=precision(x)) = set!(Arf(prec), x)
+Arb(x::BigFloat, prec::Integer=precision(x)) = set!(Arb(prec), Arf(x, prec))
+Acb(x::BigFloat, prec::Integer=precision(x)) = set!(Acb(prec), Arb(x, prec))
 
 # fallbacks:
 Arf(x::Real, prec::Integer) = set!(Arf(prec), BigFloat(x))
```

Comment:
I think we should still have a global `precision` variable as a fallback. Just if you want to be lazy and just quickly try something.

Reply:
ok, but what about thread safeness?

Reply:
Sure, this is not sensible for thread safe code, but you also do not write hastily thread safe code 

I am just thinking of defining
```
const DEFAULT_PREC = Ref(256)
precision(x::Number) = DEFAULT_PREC[]
```

In library code, one definitely should always carry around the precision